### DISCUSSION TITLE
Sort exports

### DISCUSSION THREAD

**Discussion by charmbyte** [biomejs/biome#467] [Ideas]
Is there a plan to add a feature to sort exports? I know for ESLint I would often use the "simple-import-sort" plugin to sort imports / exports. Biome replaces the import sorting well but it doesn't seem to sort exports (unless I'm missing something).

**Comment by Conaclos** [comment]
Are you aware of any tool that sort exports? This seems complex to do. How could you define an order between the exports?

**Reply by charmbyte** [reply]
The tool I've been using with ESLint is [eslint-plugin-simple-import-sort](https://github.com/lydell/eslint-plugin-simple-import-sort) . Despite its name, it includes an ESLint rule `"simple-import-sort/exports": "error"` that does [export sorting](https://github.com/lydell/eslint-plugin-simple-import-sort/blob/main/src/exports.js) .

I don't think they have covered every edge case (I may be wrong) considering it seems to be a plugin focused on sorting imports. But in every project I've used it, I haven't noticed anything concerning.

EDIT: typo

**Reply by Conaclos** [reply]
In fact, the plugin [sorts only](https://github.com/lydell/eslint-plugin-simple-import-sort#exports) `export <specifier> from <source>` declarations. This kind of declaration could be considered as an `import` followed by an `export`. In this regard, this could make sense of sorting them in the _organize-import_ action.

@biomejs/maintainers Any thought?

**Reply by victor-teles** [reply]
@Conaclos Could `organize-imports` have a new configuration option to enable this new action? To avoid introduce some unexpected behavior to the user

**Reply by charmbyte** [reply]
A quick search led me me to [eslint-plugin-sort-exports](https://github.com/jrdrg/eslint-plugin-sort-exports/blob/master/lib/sortExports.js). I've never used this plugin before, and it seems quite similar to to `eslint-plugin-simple-import-sort`, but I thought I would share it in case it provides any extra insight.

I appreciate you considering this idea. I know it's ultimately a little "fluff", but it's nice to have when needed.

**Reply by Conaclos** [reply]
@arendjr Any thoughts?

**Reply by arendjr** [reply]
I've never used something like this, but it seems like a nice to thing to have for `export ... from ...` statements indeed. Agreed it could be done the same way as how Biome organizes imports. Maybe a separate option _could_ be a good idea, but I think I would enable it by default then. It can be surprising indeed, but if I would see this happening to my code, my reaction would be "huh, neat" :)

**Reply by charmbyte** [reply]
@arendjr This is exactly how I've felt when using `simple-import-sort` with ESLint. It's 100% a  convenience feature but when I started using it with ESLint I realized how useful it really is in terms of helping the code be more organized and readable.

**Reply by charmbyte** [reply]
I'm not one to "bump", but has there been any further thought on this concept? I could take a look at it myself but my Rust skills are rudimentary at this point.

**Reply by ematipico** [reply]
@charmbyte I believe there isn't enough information to move forward, despite the initial conversations.

Maybe it's best to provide more information on where this feature could live, which code will affect, how the user could enable the feature, etc.

**Reply by atheck** [reply]
VS Codes built-in code action to organize imports also sorts exports alphabetically. I used this feature before using the biome analyzer to sort imports. But now I'm missing this feature.

I think the formatter should do this, at least if enabled in the formatter settings.

**Comment by BreakBB** [comment]
I am kind of confused that this discussion didn't get any more traction. Neither do I understand why Biome is not able to check export sorting while it can do the same for imports 🤔 

I would love to get the improved readability for exports.

**Reply by dyc3** [reply]
I believe the new import sorting also sorts exports.

**Reply by BreakBB** [reply]
First of all: What a quick response, thank you very much!

Second: I just tested the v2.0.0-beta.5 for this and you are absolutely right. Finally export sorting, looking forward to v2.0.0 🥳

**Comment by ematipico** [comment]
Closing as this is implemented and will be available in v2
